core/src/main/java/com/ttbmp/cinehub/core/usecase/buyticket/BuyTicketUseCaseController.java

```diff
@@ -0,0 +1,284 @@
+                    projectionListByMovie.add(projection);
+                }
+            }
+            List<Cinema> cinemaList = new ArrayList<>();
+            for (Projection projection : projectionListByMovie) {
+                cinemaList.add(projection.getCinema());
+            }
+            for (int i = 0; i < cinemaList.size(); i++) {
+                for (int j = i + 1; j < cinemaList.size(); j++) {
+                    if (cinemaList.get(i).getName().equals(cinemaList.get(j).getName())) {
+                        cinemaList.remove(j);
+                        break;
+                    }
+                }
+            }
+            buyTicketPresenter.presentCinemaList(new GetListCinemaResponse(CinemaDataMapper.mapToDtoList(cinemaList)));
+        } catch (Request.NullRequestException e) {
+            buyTicketPresenter.presentGetListCinemaNullRequest();
+        } catch (Request.InvalidRequestException e) {
+            buyTicketPresenter.presentInvalidGetListCinema(request);
+        }
+
+
+    }
+
+    @Override
+    public void createTicket(GetTicketBySeatsRequest request) {
+        try {
+            Request.validate(request);
+            List<Seat> seats = SeatDataMapper.mapToEntityList(request.getSeatDtoList());
+            Integer pos = request.getPos();
+            String position = request.getPosition();
+            Seat selectedSeats = seats.get(pos);
+
+            /*DECORATOR PATTERN GOF*/
+            TicketAbstract ticketAbstract = new Ticket(selectedSeats.getPrice());
+            ticketAbstract.increasePrice();
+            if (Boolean.TRUE.equals(request.getHeatedArmchairOption())) {
+                ticketAbstract = new TicketSkipLine(ticketAbstract);
+                ticketAbstract.setPrice(ticketAbstract.increasePrice());
+            }
+            if (Boolean.TRUE.equals(request.getFoldingArmchairOption())) {
+                ticketAbstract = new TicketFoldingArmchair(ticketAbstract);
+                ticketAbstract.setPrice(ticketAbstract.increasePrice());
+            }
+            if (Boolean.TRUE.equals(request.getSkipLineRadioOption())) {
+                ticketAbstract = new TicketHeatedArmchair(ticketAbstract);
+                ticketAbstract.setPrice(ticketAbstract.increasePrice());
+            }
+            /*-----------------------------------------*/
+
+            ticketAbstract.setPosition(position);
+            buyTicketPresenter.setSelectedTicket(new GetTicketBySeatsResponse(TicketDataMapper.mapToDto(ticketAbstract)));
+        } catch (Request.NullRequestException e) {
+            buyTicketPresenter.presentGetTicketBySeatsNullRequest();
+        } catch (Request.InvalidRequestException e) {
+            buyTicketPresenter.presentInvalidGetTicketBySeats(request);
+        }
+
+
+    }
+
+    @Override
+    public void getTimeOfProjection(GetTimeOfProjecitonRequest request) {
```

Comment:
This operation seems unnecessary

Reply:
It is necessary for me to retrieve the times of the projections when I am in the chooseCinema, using that method, every time I click on an element of the listView the times of that film on that date available in that cinema are loaded, unlike getProjection which is used for recover the projection of all the choices made previously

Reply:
This method has something wrong.
A projection is projected only once, but you call the presenter sending a list of projection start time.
Also, what if the client wanted to know other information besides the date of the projections?

Reply:
we will clarify this problem in a call